what4/src/What4/Expr/App.hs

```diff
@@ -937,7 +937,7 @@ instance TestEquality e => TestEquality (NonceApp t e) where
            ]
           )
 
-instance HashableF e => HashableF (NonceApp t e) where
+instance (HashableF e, TestEquality e) => HashableF (NonceApp t e) where
```

Comment:
Why is this change necessary?

Reply:
The body of this `HashableF` invokes `hashWithSalt` on fields of type `Assignment` (in the `MapOverArrays` and `FnApp` cases). In GaloisInc/parameterized-utils#127, I had to add a `TestEquality` constraint to the `Hashable` instance for `Assignment`, so that constraint must be propagated through to this `HashableF` instance as a result.